src/main/fc/core.c

```diff
@@ -804,12 +810,17 @@ bool processRx(timeUs_t currentTimeUs)
     const uint8_t throttlePercent = calculateThrottlePercentAbs();
     const bool launchControlActive = isLaunchControlActive();
 
-    if (airmodeIsEnabled() && ARMING_FLAG(ARMED) && !launchControlActive) {
-        // once throttle exceeds activate threshold, airmode latches active until disarm
+    if (ARMING_FLAG(ARMED)) {
         if (throttlePercent >= rxConfig()->airModeActivateThreshold) {
-            airmodeIsActivated = true;
+            throttleRaised = true; // Latch true until disarm
+        }
+        if (airmodeIsEnabled() && !launchControlActive) {
+            airmodeIsActivated = throttleRaised;
+        } else {
+            airmodeIsActivated = false;
```

Comment:
Can this ever happen?

Reply:
Would suggest moving line 820 before 817

Reply:
You're right, it isn't needed.